File: frontend/lib/services/invoice-service.ts
import api from "@/lib/api";
import { Invoice, PaginatedResponse } from "@/lib/types";

export const invoiceService = {
  async getMyInvoices(params?: {
    page?: number;
    size?: number;
  }): Promise<PaginatedResponse<Invoice>> {
    const response = await api.get<PaginatedResponse<Invoice>>("/invoices/my", {
      params,
    });
    return response.data;
  },

  async getAllInvoices(params?: {
    page?: number;
    size?: number;
    status?: string;
  }): Promise<PaginatedResponse<Invoice>> {
    const response = await api.get<PaginatedResponse<Invoice>>("/invoices", {
      params,
    });
    return response.data;
  },

  async getInvoiceById(id: number): Promise<Invoice> {
    const response = await api.get<Invoice>(`/invoices/${id}`);
    return response.data;
  },

  async getInvoiceByBookingId(bookingId: number): Promise<Invoice> {
    const response = await api.get<Invoice>(`/invoices/booking/${bookingId}`);
    return response.data;
  },
};

File: backend/src/main/java/com/kien/vehicle/booking/service/InvoiceService.java
package com.kien.vehicle.booking.service;

import com.kien.vehicle.booking.dto.response.InvoiceResponse;
import com.kien.vehicle.booking.dto.response.InvoiceSummaryResponse;
import com.kien.vehicle.booking.entity.Booking;
import com.kien.vehicle.booking.entity.enums.InvoiceStatus;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import java.util.List;

public interface InvoiceService {
    InvoiceResponse createInvoiceForBooking(Booking booking);
    Page<InvoiceSummaryResponse> getMyInvoices(String currentUserPhone, Pageable pageable);
    InvoiceResponse getInvoiceById(Long invoiceId, String currentUserPhone, boolean isAdmin);
    Page<InvoiceSummaryResponse>  getAllInvoices(InvoiceStatus invoiceStatus, Pageable pageable);
}

